c10/core/ScalarType.h

```diff
@@ -53,6 +53,17 @@ _(at::Half,Half,d) \
 _(float,Float,d)   \
 _(double,Double,d)
 
+#define AT_FORALL_SCALAR_TYPES(_) \
```

Comment:
I think we should try to keep some fidelity with Dispatch.h  In dispatch.h ALL doesn't include BOOL, you have to use AND_BOOL.  Can we match those here?